addons/payment_razorpay/models/payment_transaction.py

```diff
@@ -49,6 +50,18 @@ def _get_specific_rendering_values(self, processing_values):
         converted_amount = payment_utils.to_minor_currency_units(self.amount, self.currency_id)
         base_url = self.provider_id.get_base_url()
         return_url_params = {'reference': self.reference}
+
+        phone = self.partner_phone
+        if self.partner_phone:
+            # sanitize partner phone
+            country_code = self.partner_country_id.code
+            country_phone_code = self.partner_country_id.phone_code
+            phone = phone_sanitize_numbers(
+                [self.partner_phone], country_code, country_phone_code
+            ).get(self.partner_phone, {}).get('sanitized')
+            if not phone:
+                raise ValidationError("Razorpay: " + _("Incorrect phone number."))
```

Comment:
In order to get a message as precise as possible, what do you think of setting a generic error message between the line 54 and 55 for the case where there isn't a phone number set (not incorrect, just not set at all). 

Then, where you sanitize an existing number, you overwrite this message with the `msg` you got from the `phone_sanitize_numbers`?

You could then pass the resulting message to cover both the case where there isn't a phone set on the partner, and the case where it's wrong with the exact error it got. =]